test/distributed/_composable/fully_shard/test_fully_shard_util.py

```diff
@@ -112,5 +121,20 @@ def test_get_sharded_module_tree_with_module_name_to_fqns(self):
         )
 
 
+class TestUtilsSingleDevice(TestCase):
+    @pytest.mark.skipif(not is_cuda_8_9, reason="requires SM89 compatible machine")
+    def test_foreach_copy_float8(self):
+        for dtype in [
+            torch.float8_e4m3fn,
+            torch.float8_e4m3fnuz,
+            torch.float8_e5m2,
+            torch.float8_e5m2fnuz,
+        ]:
+            src = [torch.zeros(2, 2, device="cuda", dtype=dtype)] * 2
+            dst = [torch.zeros(2, 2, device="cuda", dtype=dtype)] * 2
+            # needed by fully_shard(Float8Linear)
+            torch._foreach_copy_(src, dst)
```

Comment:
nit: we might as well go one step further and actually test that the copy was correct

e.g. change `src`/`dst` to rand values and then loop over them with `torch.equal`